### PR TITLE
provider/aws: Add Security Group Rule as a top level resource

### PR DESCRIPTION
**Pull request by catsby**
First attempt at making Security Group Rules (ingress, egress) top level resources. 

This is raw, like sushi, and probably needs refinement, but it's a good start.

Missing:
- [x]  documentation 

cc @phinze

### CODE REVIEW

**Comment by geofffranks** [comment]
Any ETA on when this will be merged/released? Will fix an issue I'm running into very similar to #539 

Tried out dev binaries from this branch and it works great, just looking to see when I can expect this from upstream. Thanks!

**Comment by catsby** [comment]
cc @phinze or @mitchellh for feedback when you can

**Comment by phinze** [comment]
Code looks great!

Trying to think out the behavior here for SGs that have some rules defined nested and others not. Perhaps that's just something we solve with documentation at this point?

Basically a user who wants to use the top-level rules needs to not define any nested rules. Yeah?

(Eventually we'll build in a first-class concept of nested resources so we can express the nested ones as just "versions" of these top level ones.)

**Comment by geofffranks** [comment]
FWIW, my use case has rules defined both in the SG and in an SGR (in different modules)  and it worked great in my (albeit only one) test.

**Comment by catsby** [comment]
> Basically a user who wants to use the top-level rules needs to not define any nested rules. Yeah?

@phinze great question... I'll have to think on that and tinker with it... but I believe we'd want to solve that with documentation at this point.

**Comment by catsby** [comment]
@phinze tested it out – if a user has `egress` rules in a SG config, and also a SG_RULE resource, the two will conflict. Updates to the SG will destroy the RULE. 

Seems like Documentation is the way to "solve" this for now :frowning:

File: website/source/docs/providers/aws/r/security_group_rule.html.markdown [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+layout: "aws"
+page_title: "AWS: aws_security_group_rule"
+sidebar_current: "docs-aws-resource-security-group-rule"
+description: |-
+  Provides an security group rule resource.
+---
+
+# aws\_security\_group\_rule
+
+Provides a security group rule resource. Represents a signle `ingress` or
```
Comment (c4milo) [inline website/source/docs/providers/aws/r/security_group_rule.html.markdown]: single

**Pull request merged**